src/pip/_internal/resolution/resolvelib/candidates.py

```diff
@@ -41,7 +41,7 @@
 def make_install_req_from_link(link, parent):
```

Comment:
Would it make sense to rename `parent` to `template`/`reference`/something along those lines? `parent` sounds like it's related to the graph we're building -- parent child relationship AKA dependent-dependency relationship -- but that's not the case here.

Reply:
I think of it as `parent` in the sense that it's the req we create this one from - but yes, it does get confusing with the dependency graph. I'm happy to change, although I'm not keen on `template` - we don't take the original and "fill in the blanks" which is how I think of a template.

But a strong +1 on trying to keep addressing thses sorts of peripheral issues as we go along - it would be way too easy to end up with an accumulated pile of slightly-not-ideal decisions, and have another maintenance mess on our hands. (Says the person who inflicted PEP 517 on us 🙂)